### PR TITLE
MGMT-9545 - Failed to fetch container images needed an e2e-assisted-prod job

### PR DESCRIPTION
**Pull request by eliorerz**
Retry to start installation after some hosts fallback to ready status after the installation was already started.

/cc @osherdp @paul-maidment

### CODE REVIEW

File: src/tests/test_bootstrap_in_place.py [hunk 1]
```diff
@@ -14,15 +14,15 @@
...
-from assisted_test_infra.test_infra.utils.release_image_utils import (
-    extract_installer,
-    extract_rhcos_url_from_ocp_installer,
-)
 from assisted_test_infra.test_infra.controllers.node_controllers.node_controller import NodeController
 from assisted_test_infra.test_infra.helper_classes.config.controller_config import BaseNodeConfig
 from assisted_test_infra.test_infra.tools.assets import LibvirtNetworkAssets
 from assisted_test_infra.test_infra.utils.entity_name import ClusterName
 from assisted_test_infra.test_infra.utils.oc_utils import get_operators_status
+from assisted_test_infra.test_infra.utils.release_image_utils import (
```
Comment (eliorerz) [inline src/tests/test_bootstrap_in_place.py:22]: make reformat

File: src/assisted_test_infra/test_infra/utils/utils.py [hunk 2]
```diff
@@ -118,32 +118,6 @@ def set_tfvars(tf_folder, tfvars_json):
         json.dump(tfvars_json, _file)
 
 
-def are_hosts_in_status(hosts, nodes_count, statuses, status_info="", fall_on_error_status=True):
```
Comment (eliorerz) [inline src/assisted_test_infra/test_infra/utils/utils.py:121]: Not used here and on QE projects

File: src/assisted_test_infra/test_infra/utils/release_image_utils.py [hunk 3]
```diff
@@ -1,5 +1,6 @@
-import logging
 import json
+import logging
```
Comment (eliorerz) [inline src/assisted_test_infra/test_infra/utils/release_image_utils.py:2]: make reformat

**Comment by osherdp** [comment]
/hold
(sorry about that, just wanted to pause and rethink about it)

What was the reason cluster regressed from ``prepare-for-installation`` to ``ready``?
If it's the image-pull validation, then I think it's problematic that we don't retry it at all (at least from what I've been told). Maybe the right thing to do here is to add 2 retries to this action before regressing the state

I'm suggesting it because retrying is usually adding some noise that we would like to eliminate, plus the UX might be improved

File: src/service_client/logger.py [hunk 4]
```diff
@@ -88,20 +88,22 @@ def add_stream_handler(logger: logging.Logger):
     logger.addHandler(ch)
 
 
-def get_logger(name: str = ""):
-    logging.getLogger("requests").setLevel(logging.ERROR)
-    logging.getLogger("urllib3").setLevel(logging.ERROR)
+logger_name = os.environ.get("LOGGER_NAME", "")
+urlib3_logger = logging.getLogger("urllib3")
+urlib3_logger.handlers = [logging.NullHandler()]
```
Comment (osherdp) [inline src/service_client/logger.py:93]:
```suggestion
urllib3_logger = logging.getLogger("urllib3")
urllib3_logger.handlers = [logging.NullHandler()]
```

File: src/service_client/logger.py [hunk 5]
```diff
@@ -88,20 +88,22 @@ def add_stream_handler(logger: logging.Logger):
...
 
-    logger = logging.getLogger(name)
-    logger.setLevel(logging.DEBUG)
+logging.getLogger("requests").setLevel(logging.ERROR)
+urlib3_logger.setLevel(logging.ERROR)
```
Comment (osherdp) [inline src/service_client/logger.py:96]:
```suggestion
urllib3_logger.setLevel(logging.ERROR)
```

File: src/service_client/logger.py [hunk 6]
```diff
@@ -88,20 +88,22 @@ def add_stream_handler(logger: logging.Logger):
...
-    add_log_file_handler(logger, "test_infra.log")
-    add_stream_handler(logger)
-    return logger
+_logger = logging.getLogger(logger_name)
+_logger.setLevel(logging.DEBUG)
 
+add_log_file_handler(_logger, "test_infra.log")
+add_log_file_handler(urlib3_logger, "test_infra.log")
+add_stream_handler(_logger)
+add_stream_handler(urlib3_logger)
```
Comment (osherdp) [inline src/service_client/logger.py:104]:
```suggestion
add_log_file_handler(urllib3_logger, "test_infra.log")
add_stream_handler(_logger)
add_stream_handler(urllib3_logger)
```

File: src/service_client/logger.py [hunk 7]
```diff
@@ -88,20 +88,22 @@ def add_stream_handler(logger: logging.Logger):
...
 
-    logger = logging.getLogger(name)
-    logger.setLevel(logging.DEBUG)
+logging.getLogger("requests").setLevel(logging.ERROR)
+urlib3_logger.setLevel(logging.ERROR)
 
-    add_log_file_handler(logger, "test_infra.log")
-    add_stream_handler(logger)
-    return logger
+_logger = logging.getLogger(logger_name)
```
Comment (osherdp) [inline src/service_client/logger.py:98]: what's the reason for working with ``_logger`` as a temporary variable?
please add a comment to explain that if there's any specific issue with the straight-forward solution

**Review by osherdp — CHANGES_REQUESTED** [review]

**Review by osherdp — APPROVED** [review]

**Comment by osherdp** [comment]
/hold
(waiting for the optional job ``e2e-metal-assisted-prod``

**Comment by eliorerz** [comment]
@osherdp @tsorya We got lucky, We had the `Failed to fetch container images needed for installation` error on this PR.
The test was able to restart the installation and finish successfully.

`2022-03-09 16:01:47,199 - root - WARNING - 140272010113280 - An error was occurred during the installation process that caused the cluster to fallback to ready status. Retrying (Attempted left 2) ...`

**Pull request merged**